### ISSUE TITLE
'target=...' changes break some compiler-rt tests on sparcv9

### ISSUE THREAD

**Issue by rorth** [llvm/llvm-project#60002]
Since the `target='...'` patches landed, some `compiler-rt` tests `XPASS` on Solaris/sparcv9:
```
  XPASS: SanitizerCommon-ubsan-sparcv9-SunOS:: printf-ldbl.c
  XPASS: SanitizerCommon-ubsan-sparcv9-SunOS:: scanf-ldbl.c
  XPASS: UBSan-Standalone-sparcv9::cast-overflow.cpp
  XPASS: UBSan-Standalone-sparcv9::log-path_test.cpp
```
which is a regression.  These were all changed like this:
```
-// XFAIL: sparc-target-arch && solaris
+// XFAIL: target={{sparc.*solaris.*}}
```
i.e. previously they were **only** `XFAIL`ed for 32-bit Solaris/sparc, while now the `XFAIL` applies unconditionally.  The `target='...'` keyword uses `target_triple` , which is **always** `sparcv9-sun-solaris2.11` in this build, both for tests built with `-m64` as well as for those built with `-m32`.  This is not a Solaris issue at all, but happens e.g. on `x86_64-pc-linux-gnu` (i.e. everywhere).

There are two options here:
- Those parts of the patch are wrong and should be reverted.  While this would restore the status quo, this doesn't really seem desirable in the long term if the goal is to use the `target='...'` syntax universally.
- If things should be expressible with `target='...'`, `target_triple` needs to be set differently for (e.g.) 32 and 64-bit multilibs, which is likely a massive change.

**Comment by pogo59** [comment]
Thanks for the report.  It sounds like the 32-bit case sets `sparc-target-arch` but the 64-bit case sets `somethingelse-target-arch`, is that correct? According to compiler-rt\test\lit.common.cfg.py this feature keyword is based on target_arch, which is set by COMPILER_RT_DEFAULT_TARGET_ARCH in lit.common.configured.in.

If that's the case, then changing the condition to
`XFAIL: sparc-target-arch && target={{.*solaris.*}}`
should fix it.  I'll put up a patch, and if you could verify it that would be lovely.

**Comment by pogo59** [comment]
I don't see a "rorth" in Phabricator, so I'll just commit the change. Please let me know whether it works.

**Comment by rorth** [comment]
> I don't see a "rorth" in Phabricator, so I'll just commit the change. Please let me know whether it works.

I'm "ro" over threre (probably even before I got onto github).  I'll just let the Solaris/sparcv9  buildbot do it's thing.  Thanks for the quick fix.

**Comment by pogo59** [comment]
Oops had to fix an embarrassing typo in 16ee7c49

**Comment by rorth** [comment]
Once the Solaris/sparcv9 buildbot had caught up, results were fine again.  Thanks for the quick fix.
